entry/src/entry.rs

```diff
@@ -44,11 +43,13 @@ use {
     },
 };
 
-thread_local!(static PAR_THREAD_POOL: RefCell<ThreadPool> = RefCell::new(rayon::ThreadPoolBuilder::new()
-                    .num_threads(get_thread_count())
-                    .thread_name(|ix| format!("entry_{}", ix))
-                    .build()
-                    .unwrap()));
+lazy_static! {
+    static ref PAR_THREAD_POOL: ThreadPool = rayon::ThreadPoolBuilder::new()
+        .num_threads(num_cpus::get())
```

Comment:
Changed this from `get_thread_count()` to `num_cpus::get()` to match current number of threads on master.
Apparently this is initialized from two separate threads.

Reply:
intentionally doubling the thread count and disregarding the `SOLANA_RAYON_THREADS` envvar?

Reply:
I don't want to change number of threads in each thread-pool in this commit.
This matches the number of threads for this task with the current master code.

Reply:
`get_thread_count()` is `num_cpus::get() / 2` though, no?

https://github.com/solana-labs/solana/blob/b4c3b66f1a835f2dee4e826cf7b652b545069f60/rayon-threadlimit/src/lib.rs#L7-L18

Reply:
> `get_thread_count()` is `num_cpus::get() / 2` though, no?

yes, but apparently two separate threads are initializing this thread-local thread-pool.
so we have 2 thread-pools each with n-cpu/2 threads; so n-cpu threads in total.
This commit replaces those 2 thread-pools with only one but with same number of threads as the two before.

Reply:
Ah, I see.  I'd misinterpreted that latter comment

Reply:
I guess the question about ignoring the envvar still stands. I think some validators are using it, though I'm not sure whether to any practical affect

Reply:
another option is to change the number of threads here to `get_thread_count() * 2` to:
* keep the size of thread-pools the same as current master
* keep the thread-pool size proportional to envvar.

would that make sense?

Reply:
probably unintuitive to double an envar-specified value.  tfw the helper is actually hurting you :face_with_head_bandage: 

would a second helper that doesn't divide by two in the default case be reasonable?

Reply:
any suggestion to what name the new helper function? `get_thread_count` vs `...`?

Reply:
https://github.com/solana-labs/solana/blob/b4c3b66f1a835f2dee4e826cf7b652b545069f60/rayon-threadlimit/src/lib.rs#L8-L9
`get_thread_count_whole_hog`? :joy: 

How about something like `get_max_thread_count`?

Reply:
done